pygac/tests/test_corrections.py

```diff
@@ -0,0 +1,227 @@
+        pass
+
+
+class TestCorrections(unittest.TestCase):
+    def setUp(self):
+        # Create artificial scanline numbers
+        along_track = 12000
+        scans = np.zeros(12000, dtype=[("scan_line_number", ">u2")])
+        scans["scan_line_number"] = np.arange(1, along_track+1)
+
+        # ... with 200 missing scanlines at scanline 8000
+        scans["scan_line_number"][8000:] += 500
+        self.correct_scan_line_numbers = scans["scan_line_number"].copy()
+
+        # ... and some spikes here and there
+        scans["scan_line_number"][3000] += 1E4
+        scans["scan_line_number"][9000] -= 1E4
+        self.correct_scan_line_numbers = np.delete(
+            self.correct_scan_line_numbers, [3000, 9000])
+
+        # Create artificial timestamps
+        scanning_rate = 2.0 / 1000.0  # scanlines per millisecond
+        t0 = np.array([TEST_TIMESTAMP.isoformat()],
+                      dtype="datetime64[ms]").astype("i8")[0]
+        shift = 1000
+        msecs = t0 + shift + self.correct_scan_line_numbers/scanning_rate
+        self.correct_utcs = msecs.copy().astype(">M8[ms]")
+
+        # ... with some corruptions here and there
+        msecs[1000] += 24*3600*1000
+        msecs[2000] -= 24*3600*1000
+        msecs[3000:] += 1800*1000
+
+        # Create a GAC Reader and parse the test data
+        self.reader = TestGACReader()
+        self.reader.filename = 'test'
+        self.reader.scans = scans
+        self.reader.utcs = msecs.astype(">M8[ms]")
+        self.reader.times = self.reader.utcs.astype(datetime.datetime)
+
+        # Create artificial channel data with some noisy regions. Channel order:
+        # 1, 2, 3b, 4, 5, 3a
+        cols = np.arange(409)
+        rows = np.arange(12000)
+        mcols, mrows = np.meshgrid(cols, rows)
+        data = np.sin(mrows/float(rows.size))*np.cos(mcols/float(cols.size))
+        channels = [data.copy() for i in range(6)]
+
+        # ... add noise to channel 1 and 4
+        channels_noise = [c.copy() for c in channels]
+        noise_cols, noise_rows = np.meshgrid(np.arange(200, 300),
+                                             np.arange(4000, 5000))
+        ampl = 0.5
+        for ichan in (0, 3):
+            channels_noise[ichan][noise_rows, noise_cols] += ampl*np.random.rand(
+                noise_rows.shape[0], noise_rows.shape[1])
+
+        # ... scale data
+        for i in (0, 1, 5):
+            # Reflectances
+            channels[i] /= 0.01*0.01
+            channels_noise[i] /= 0.01 * 0.01
+        for i in (2, 3, 4):
+            # Brightness temperatures
+            channels[i] = (channels[i] - 273.15) / 0.01
+            channels_noise[i] = (channels_noise[i] - 273.15)/0.01
+
+        self.channels = channels
+        self.channels_noise = channels_noise
+        self.noise_cols = noise_cols
+        self.noise_rows = noise_rows
+
+    def test_scan_line_number(self):
+        """
+        Test whether scan line number correction works as expected
```

Comment:
@Funkensieper For reference: https://www.python.org/dev/peps/pep-0257/#one-line-docstrings